api/utils/exceptions.py

```diff
@@ -0,0 +1,38 @@
+from math import ceil
+from fastapi import status
+from fastapi import HTTPException, Request, Response
+from fastapi.responses import JSONResponse
+from fastapi.utils import is_body_allowed_for_status_code
+
+from api.core import responses
+
+
+async def http_exception_handler(request: Request, exc: HTTPException) -> Response:
+    headers = getattr(exc, "headers", None)
+    if not is_body_allowed_for_status_code(exc.status_code):
+        return Response(status_code=exc.status_code, headers=headers)
+    return JSONResponse(
+        {
+            "message": exc.detail,
+            "success": False,
+            "status_code": exc.status_code
+        }, 
+        status_code=exc.status_code, headers=headers
+    )
+
+
```

Comment:
This is nice, as exceptions should be raised where necessary. One thing though, check out the `JsonResponseDict` class in the `json_response` module